patches/api/0416-Add-whitelist-events.patch

```diff
@@ -0,0 +1,146 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: SageSphinx63920 <sage@sagesphinx63920.dev>
+Date: Sun, 14 May 2023 12:56:15 +0200
+Subject: [PATCH] Add whitelist events
+
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerWhitelistAddEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerWhitelistAddEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..891df521eec218741f7200003a3112a66c9b10c0
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerWhitelistAddEvent.java
+@@ -0,0 +1,64 @@
++package io.papermc.paper.event.player;
++
++import com.destroystokyo.paper.profile.PlayerProfile;
++import org.bukkit.Bukkit;
++import org.bukkit.OfflinePlayer;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.Event;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when a player is being added to the whitelist
++ */
++public class PlayerWhitelistAddEvent extends Event implements Cancellable {
```

Comment:
Should this maybe be a ServerEvent?

Reply:
I think updates regarding the player should be a PlayerEvent_

Reply:
Well it still isn't using the PlayerEvent, and I'm not sure that it's applicable because the Player object might not exist, for an offline player.

Reply:
There are other events in the player package not using the player event

Reply:
Maybe just call it something like `WhitelistModifyEvent` then?

Reply:
It can't extend PlayerEvent because nullability, server would maybe be a better place because it's more concerned about the state of a system within the server; don't think it's really a profile event, but, maybe?

Reply:
I think calling it WhitelistStateUpdateEvent would be a good name in the Server package